runtime/service.go

```diff
@@ -598,44 +600,65 @@ func (s *service) startVM(ctx context.Context, request *taskAPI.CreateTaskReques
 	}
 
 	cfg := firecracker.Config{
-		BinPath:         s.config.FirecrackerBinaryPath,
 		SocketPath:      s.config.SocketPath,
 		VsockDevices:    []firecracker.VsockDevice{{Path: "root", CID: cid}},
 		KernelImagePath: s.config.KernelImagePath,
 		KernelArgs:      s.config.KernelArgs,
-		RootDrive:       firecracker.BlockDevice{HostPath: s.config.RootDrive, Mode: "rw"},
-		CPUCount:        int64(s.config.CPUCount),
-		CPUTemplate:     firecracker.CPUTemplate(s.config.CPUTemplate),
-		MemInMiB:        256,
-		Console:         s.config.Console,
-		LogFifo:         s.config.LogFifo,
-		LogLevel:        s.config.LogLevel,
-		MetricsFifo:     s.config.MetricsFifo,
-		Debug:           s.config.Debug,
-	}
+		MachineCfg: models.MachineConfiguration{
+			VcpuCount:   int64(s.config.CPUCount),
+			CPUTemplate: models.CPUTemplate(s.config.CPUTemplate),
+			MemSizeMib:  256,
+		},
+		LogFifo:     s.config.LogFifo,
+		LogLevel:    s.config.LogLevel,
+		MetricsFifo: s.config.MetricsFifo,
+		Debug:       s.config.Debug,
+	}
+
+	idx := strconv.Itoa(1)
+	cfg.Drives = append(cfg.Drives,
+		models.Drive{
+			DriveID:      &idx,
+			PathOnHost:   &s.config.RootDrive,
+			IsRootDevice: firecracker.Bool(true),
+			IsReadOnly:   firecracker.Bool(false),
+		})
 
 	// Attach block devices passed from snapshotter
-	for _, mnt := range request.Rootfs {
+	for i, mnt := range request.Rootfs {
 		if mnt.Type != supportedMountFSType {
 			return nil, errors.Errorf("unsupported mount type '%s', expected '%s'", mnt.Type, supportedMountFSType)
 		}
-
-		cfg.AdditionalDrives = append(cfg.AdditionalDrives, firecracker.BlockDevice{HostPath: mnt.Source, Mode: "rw"})
-	}
-
-	s.machine = firecracker.NewMachine(cfg, firecracker.WithLogger(log.G(ctx)))
-	s.machineCID = cid
-
-	log.G(ctx).Println("initializing machine")
-	if _, err := s.machine.Init(ctx); err != nil {
-		log.G(ctx).WithError(err).Error("machine init failed")
+		idx := strconv.Itoa(i + 2)
+		root := false
+		ro := false
+		cfg.Drives = append(cfg.Drives,
+			models.Drive{
+				DriveID:      &idx,
+				PathOnHost:   firecracker.String(mnt.Source),
+				IsRootDevice: &root,
+				IsReadOnly:   &ro,
```

Comment:
Can these two be just inlined? 
```go
IsRootDrive: firecracker.Bool(false)
IsReadOnly:  firecracker.Bool(false)
```

Reply:
Yes, they should be inlined.

Beyond that, though, the client shouldn't need to care. These should also be handled within the SDK...